### ISSUE TITLE
dart layer,  spacemacs freezing when edit .dart file , flutter 

### ISSUE THREAD

**Issue by TPLinux** [syl20bnr/spacemacs#12820]
Hi i am trying to code flutter/dart using Spacemacs

when i added layer 'dart' to layers , all sounds good even the dart language server working ..

but when i edit the file , Spacemacs freezing and finally give me this message 

> completion--some: Timeout while waiting for response. Method: textDocument/completion

**Comment by duianto** [comment]
Do you have some reproduction steps?

And could you provide your system info, press `SPC h d s` to copy it to your clipboard.

**Comment by TPLinux** [comment]
> Do you have some reproduction steps?
> 
> And could you provide your system info, press `SPC h d s` to copy it to your clipboard.

#### System Info :computer:
- OS: darwin
- Emacs: 26.2
- Spacemacs: 0.200.13
- Spacemacs branch: master (rev. 26b8fe0c3)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((python :variables python-backend 'anaconda python-sort-imports-on-save t python-test-runner 'pytest)
 javascript html helm auto-completion emacs-lisp git spell-checking syntax-checking shell)
```
- System configuration features: NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS

**Comment by TPLinux** [comment]
i think there is a problem with lsp mode , even with python

**Comment by duianto** [comment]
I don't think there has been any lsp related commits on the `master` branch.
They will be included when the next version is released. I don't have any insight into when that is.

**Comment by itome** [comment]
`dart_language_server` that used for lsp backend, is very slow and now obsoleted.
I'll make PR to change lsp backend to dart analysis server which is official dart lsp backend. Maybe it makes some improvements in performance and stability.

**Comment by nicobao** [comment]
Hi!
I am not completely sure it is related. But since I use `lsp` and the `dart` layer together for editing .dart files, Spacemacs periodically freezes. I have to kill emacs... @itome what could I do as a workaround to use the "dart analysis server" instead of "dart_language_server", before you send your PR?

**Comment by yyoncho** [comment]
@nicobao follow https://github.com/emacs-lsp/lsp-mode#reporting-performance-problems 

Generally, emacs 26.3 won't work fine with dart language server. Probably setting lsp-dart-suggest-from-unimported-libraries might help but for acceptable performance, you need Emacs27+ native json parsing.

**Comment by nicobao** [comment]
Thanks, I upgraded to Emacs 27.0.91 and it works neat.
I faced this issue: https://github.com/emacs-lsp/lsp-mode/issues/1606 
after installation of Emacs 27 but I solved it by adding `lsp-dart` to the additional packages as suggested in the comments. Hopefully after https://github.com/syl20bnr/spacemacs/pull/13518 is merged, no error will pop up.

**Comment by yyoncho** [comment]
@nicobao nice, maybe close the issue to keep spacemacs backlog shorter?

**Comment by ericdallo** [comment]
Also Dart LSP server was returning a lot of completion items and this was fixed in dart-sdk `2.9.0-19.0`, more info [here](https://github.com/dart-lang/sdk/issues/40026#issuecomment-650046508).

I think we can close this issue.
